kinoml/ml/torch_models.py

```diff
@@ -179,28 +183,182 @@ def __init__(
         self.fully_connected_out = nn.Linear(self.hidden_shape, self.output_shape)
 
     def forward(self, x):
         """
         Defines the foward pass for a given input 'x'
         """
+        x = x[0]
+        x = x.float()
         x = self._activation(self.convolution(x))
         x = torch.flatten(x, 1)
         x = self._activation(self.fully_connected_1(x))
         return self.fully_connected_out(x)
+
+class ConvolutionNeuralNetworkRegressionKinaseInformed(_BaseModule):
+    """
+    Builds a Convolutional Neural Network and a feed-forward pass for a kinase-ligand setting.
+
+    Parameters
+    ----------
+    input_shape : tuple
+        Dimension of input tensors, for the ligand and the protein.
+    embedding_shape : int, default=200
+        Dimension of the embedding after convolution.
+    kernel_shape : int, default=10
+        Size of the kernel for the convolution.
+    hidden_shape : int, default=100
+        Number of units in the hidden layer.
+    output_shape : int, default=1
+        Size of the last unit, representing delta_g_over_kt in our setting.
+    activation : torch function, default=relu
+        The activation function used in the hidden (only!) layer of the network.
+    """
+
+
+    def __init__(
+        self,
+        input_shape,
+        embedding_shape=300,
+        kernel_shape=10,
+        hidden_shape=100,
+        output_shape=1,
+        activation=F.relu,
+    ):
+        super().__init__()
+
+        self.input_shape = input_shape
+        # Ligand shape
+        # nb_char = self.input_shape[0][0],
+        # max_length_smiles = self.input_shape[0][1],
+
+        # Protein shape
+        # nb_residues = self.input_shape[1][0],
+        # max_length_seq = self.input_shape[1][1],
+
+        self.embedding_shape = embedding_shape
+        self.kernel_shape = kernel_shape
+        self.hidden_shape = hidden_shape
+        self.output_shape = output_shape
+        self._activation = activation
+
+        # Convolution on ligand
+        self.convolution_ligand = nn.Conv1d(
+            in_channels = self.input_shape[0][0],
+            out_channels=self.embedding_shape,
+            kernel_size=self.kernel_shape,
+        )
+        self.temp_ligand = (self.input_shape[0][1] - self.kernel_shape + 1) * self.embedding_shape
+
+        # Convolution on protein
+        self.convolution_protein = nn.Conv1d(
+            in_channels=self.input_shape[1][0],
+            out_channels=self.embedding_shape,
+            kernel_size=self.kernel_shape,
+        )
+        self.temp_protein = (self.input_shape[1][1] - self.kernel_shape + 1) * self.embedding_shape
+
+        self.fully_connected_1 = nn.Linear(self.temp_ligand + self.temp_protein, self.hidden_shape)
+        self.fully_connected_out = nn.Linear(self.hidden_shape, self.output_shape)
+
+    def forward(self, x):
+        """
+        Defines the foward pass for given an input composed of two entities: ligand and protein.
+        """
```

Comment:
Can we document the expected shape and content of `x` here?

Reply:
Adapted docstrings accordingly.